python-avd/pyavd/_eos_designs/structured_config/underlay/router_ospf.py

```diff
@@ -37,14 +37,25 @@ def router_ospf(self: AvdStructuredConfigUnderlay) -> dict | None:
             mlag_l3_vlan = default(self.shared_utils.mlag_peer_l3_vlan, self.shared_utils.mlag_peer_vlan)
             no_passive_interfaces.append(f"Vlan{mlag_l3_vlan}")
 
-        process = {
-            "id": process_id,
-            "passive_interface_default": True,
-            "router_id": self.shared_utils.router_id,
-            "max_lsa": get(self._hostvars, "underlay_ospf_max_lsa", default=12000),
-            "no_passive_interfaces": no_passive_interfaces,
-            "bfd_enable": get(self._hostvars, "underlay_ospf_bfd_enable", default=False),
-        }
+        process = {}
+
+        process.update(
+            {
+                "id": process_id,
+                "passive_interface_default": True,
+            }
+        )
+
+        if self.shared_utils.use_router_general_for_router_id is False:
+            process["router_id"] = self.shared_utils.router_id
+
+        process.update(
+            {
+                "max_lsa": get(self._hostvars, "underlay_ospf_max_lsa", default=12000),
+                "no_passive_interfaces": no_passive_interfaces,
+                "bfd_enable": get(self._hostvars, "underlay_ospf_bfd_enable", default=False),
+            }
+        )
```

Comment:
Please move this to the first process. No need to use update multiple times. You are already reordering the output for other structured config so might as well do it here too. Another option is to revert to the original and then do `self.shared_utils.router_id if not self.shared_utils.use_router_general_for_router_id else None`

Reply:
Fixed.